Jam Map Changes v5 - Hightown Edition (again)

Jamdrawers opened the pull request:
These changes are all a bit more experimental but with the precariousness of things lately I figure we might as well.


1) Rejiggering the kitchen layout so that you can enter the dungeons without a kitchen key or going into the dungeon, the servant quarters aren't right behind the throne so you can RP there without the whole court hearing you, and the food serving area being closer to the throneroom/visitors. Also sections off the basement entrance so that it doesn't require a kitchen key.
![4763546c8dc2e3618bf0bb659d4b74b4](https://github.com/user-attachments/assets/711f1e2f-7e7d-4297-b990-6f16de10adca)

A little cloak addition to the Lord's bedroom along with a minor aesthetic touch-up. Felt odd that there were only women's clothes in that closet.
![60cdfced0a3c3af59bba21b9726d5231](https://github.com/user-attachments/assets/5ce98300-08f8-44ec-896e-26d7f62acbe1)

The Merchant's:
Added some randomised bits of gear spawning in the merchant's building. Just some stuff for you to try to sell, a random surprise. Might break up the monotony of everything being goldface-bought to order.
![a59a74353d14d0848fbc089474c562c8](https://github.com/user-attachments/assets/c58d27bc-2934-47d7-9070-07ec1ad5b3e9)

The Whitevein!


More silly outfits for the nightmaiden to use - the perfect place for you to find a studded leather bikini
![851b3d4ebb3a5d0cffa1cc77faf746ca](https://github.com/user-attachments/assets/d6da1b97-f508-4335-b26f-eb0ee20d446b)

Stock of drugs, alcohol and roguery supplies in the whitevein store
![b4d1db6a31a949a3a6367514e838aee0](https://github.com/user-attachments/assets/b3c25510-9c7b-4ce5-91af-5ec8b95a99eb)

Widening out the floor layout to be a bit more open. Replaced the bad massage parlours that nobody used with comfy little nooks. Trimmed down the dormitories (we don't need 10 beds for the whores and henchmen). Will ideally be a bit more inviting and lead people into the best rooms.
![96cadd9027b6ce906ac2d9694e1be3a1](https://github.com/user-attachments/assets/2b3d5017-cd6d-43b1-8662-1a7e440112a7)

Would be worth a touch up if/when the AP nightmaster goldface stuff gets in
And if some of the associated sprites/tiles get added too, such as cushions and curtains
Would also be worth doing another pass after adding a new series of keys for both individual hireable rooms similar to the tavern, and also having a unique key for the nightmaster and not all the nightmaidens. Brought in a little secret stash for the nightmaster full of drugs and rogue tools

Biggest change - moved the arena over here, connected to the Whitevein with the main doors and closets using Whitevein keys. And it readds the Fight-pit-viewing point! Again this is kind of experimental, I could go back on the viewing port or indeed the whole thing, buuuuut it could be fun! It'd probably make sense to put cage walls around it but in the short term I think it would be fun to have the possibility of throwing someone down there. I made sure the gate doors in the arena are set to open by default, so when not in use it you generally shouldn't get stuck there - and even if you do, you're in prime screaming distance for passerbyes.
It has half-hidden doors connecting to the sewers, so a sewer route could cut right through it
I had to redirect the sewers a little to accomodate it, and I had to shift and shrink the pop-up shop north of the merchants a smidge to make the sewer hold next to it it still drop into water.

I also added another public utility mail booth/notice board in front of it. I found it was kind of difficult to access a postbox that isn't behind a locked door - even the tavern one you have to jump over the counter and clamber over a rack of mugs.  The only really public one is right at the gate which feels like an awkward place to do paperwork. I'm snipping that one out - 'Bastion's Entrance' is a confusing name (bastion? Which bastion? We have 3 things that get called Bastions), and the new public notice board is close enough anyway and hopefully a more interesting and interactive place to be led to. Also might encourage people to leave public notes.
![3b257b294e6b6c683ca460523f5d3793](https://github.com/user-attachments/assets/2c9f95bc-09f8-4ccf-97ab-99c9323702a6)

![c44ff44ab8e3a6b21dde87c67745a4a0](https://github.com/user-attachments/assets/be077f49-6601-4bb8-ba70-91e394acc30c)

In the location where the arena *was*, I added a small basement to the guild smith for just a little extra storage space. (with a shuttered back exit to make it reasonably difficult to break into and adjusted the smithy itself just a little to accomodate the stairs and clean up the floor layout a little
![9a03abd2e470a4685906b1b21b235c9d](https://github.com/user-attachments/assets/3942ca30-7119-4d6b-8095-c4fdfa17997d)
![651be3df7ec820fc0eee03893740736f](https://github.com/user-attachments/assets/986918b3-683b-4376-aa20-f4ad2c2c2d2f)


![image](https://github.com/user-attachments/assets/c961fa29-a49d-43d0-8ff6-55e78229a4bf)
Changed the ported lavamines to *mostly* have lavamine-quality ore. Makes it a little more consistent

![image](https://github.com/user-attachments/assets/4a50888c-6123-43a4-9171-fc5751a7b3c5)
Visual change to the independent merchant, differentiates it from the regular merchant

![image](https://github.com/user-attachments/assets/05db908b-f44d-482a-98ac-994f60621ee3)
Shovel in the graveyard!


Adjustments to the church basement: The left side of the dungeon uses church keys, meaning that you don't need an inquisitor or priest to be able to use the regular cell, only the advanced interrogation rooms and inquisitorial chambers. One of the cells, a bigger one, has no windows. Which must be boring for whoever gets put in there... but maybe sometimes no windows is more evil! So why not have Shutters so you can decide whether your prisoner gets daylight or not.

I decorated the inquisitor's quarters a little just to be less bland, and made the weird secret exit go out of the inquisitors quarters that they can use it as a secret exit. Also comes with an extra cell kept close to the interrogation room for convenience, only openable with the inquisitor's key, so an inquisitor doesn't have to worry about sympathetic eorans letting heretics go free.
Oh, also, threw some Mana Binding Shackles into the inquisitor's stuff. It feels like exactly the sort of thing an inquisitor should have. Fixed a typo as well.

![cbdba03827a4b41e653dbbe1cd0ca72b](https://github.com/user-attachments/assets/ab3df082-2d71-4bde-b6eb-43072069be89)

New location: Inquisitorial Safehouse. Hidden in the most remote and least desirable towner house, the confessional key leads to a tunnel that leads to a basement that leads to a little safehouse with a bedroom, storeroom, prison and tortureroom. All pleasingly by the river, offering easy access to body disposal and/or daring escapes.
![29817ddc430e7addc0b2fe9f5c0e7d4d](https://github.com/user-attachments/assets/e1c303f7-fff7-4441-94a4-a1173c908117)
![5ad67091b41b0316b09c457108beccd1](https://github.com/user-attachments/assets/b03b67ff-d9d2-4a0d-9d49-b1567d5107ba)
![27b88bf089f3975bc0180307afc260cc](https://github.com/user-attachments/assets/8acaf2bf-eb8e-467b-af2a-a3779271f98e)
![f2ed1d964d5e71f3b7e4a674923c74e0](https://github.com/user-attachments/assets/03df7ead-cedd-4f1f-874c-f7ba3fdf19d9)

A little secret area, a hole in the sewers leading to the whitevein's bath outflow with a little hobo camp you can swim by.
![747fb2a8da1ef4f54706af3686b27aad](https://github.com/user-attachments/assets/cc7ce2f2-f271-4541-bab1-32ee8821cbe7)
![0c7760ec951a4d995fae32bf3a142d5f](https://github.com/user-attachments/assets/4f61141e-4be6-49e7-961d-0c1c57248293)

Next, the watchman building. I don't have strong opinions about its functionality so I mostly just changed the aesthetics to make it a little more cramped and a little less bland with some windows on overhands to keep them reasonably secure. I notice it's one of the least changed buildings since Old-Rockhill and also as such one of the ugliest. I threw a few light gambesons and cudgels into the armoury too, why not. Good to have some cheap backups, but not so good that they don't leave easy room for upgrades. Then added in some more desks and papers for the important paperwork, partly for use and partly for decoration.
![0ccfd379efece52b917eec3e87c41959](https://github.com/user-attachments/assets/109dc1f9-5109-4a18-89f6-6c11817caa28)

On the first floor, I rearranged things to make it a little less open and empty with the chests being in little nooks and the door inset a bit to give it more character, and added barred stone windows to the cell here. If you want a more secure, private, windowless cell, you already have a few downstairs - I thought it'd be nice to have more of a drunk-tank where you can have prisoners heckle or be heckled or be pelted with rotten fruit - or weapons and lockpicks if someone is trying to help you escape!
![0ccfd379efece52b917eec3e87c41959](https://github.com/user-attachments/assets/95c78d8e-3008-4f39-aa5d-69dd68b317b4)
At the request of a player I added a little booth with a paperwork table where one might trade notes or ask for a watchman to help them with something.
![ec8065c78d16d5d25aea501a95b97a08](https://github.com/user-attachments/assets/6da99f99-55c6-4063-b402-80a40cb7cd91)

A more experimental change I could go either way on - a second counter on the side of the seamsters to potentially encourage a bit more commerce and make them more visible, with the backdoor moved to the alleyway. Not a huge deal.
![a9b2450d9fbe282495da1f3e980d9d53](https://github.com/user-attachments/assets/c87b07ab-f7cf-4d55-a460-499eee0f652e)


I did a little redocrating around the bandit camp, not really touching usability but just making it a bit less plain and square and a bit more neat and comfy, putting beds into little cubbyholes and varying up the textures. Also adds torches - was the bandit camp so old it was built and not touched since before they added wall-mounted torches?
![ab569defba49e891f49d5f9efa09abaf](https://github.com/user-attachments/assets/732fe961-616d-496c-a179-092d975469eb)

Made the vanguard gate closed by default. Just a little reminder that you have a gate you can use - and makes it less easily smashable when the vanguard aren't about.
![82cda7838cd4da20abe1271da1a64c16](https://github.com/user-attachments/assets/567d4b2f-e391-467c-b2d8-c2201551c9b6)

Tunnel from butcher's into soiler's. Partly for convenience, partly because it's neat, partly for scary surprises.
![708aee952f1f0e0d0bf02931beac6a8f](https://github.com/user-attachments/assets/df7a1d12-6522-4572-b16a-40e8972c329a)

Stairs in the Gatehouse going up to make access to the increased storage space just a little more ergonomic.
![Uploading a83eaa26
![29e0e9c3ecc0b01522290fb346032fcc](https://github.com/user-attachments/assets/710db8e0-be3e-413b-86c1-4eafab749d37)
d6ea588f5df9636abe6e6d70.png…]()

There was a towner house with a garden in the back. I flipped the garden to be visible from the main-road rather than hidden behind the town.
![d9f7599f78ade7159353fc5df55fb62c](https://github.com/user-attachments/assets/a34ac2fa-cd0b-4bb3-974a-5f95f9e212ce)



## Why It's Good For The Game

I think I described why it's good one by one after each example. My last set of changes seemed pretty well recieved. I love Rockhill so much...


## Proof of Testing (Required)


![43cefa505950904c1bd695f35f95ce6b](https://github.com/user-attachments/assets/6cf1caa2-f214-4a45-9cc3-82311e00d69e)
I ran through it three or four times in visual studio, so I *think* I've caught all walled doors or nonfunctioning levers or unlit areas; the screenshots above aren't all up-to-date.

RadiantFlash7 reviewed (changes requested):
Two Major things:
No saiga in the keep round start. We don't want keep players to be riding saiga around into combat off the bat. This is something I'm VERY firm on.

Secondly, and this is a lot more up for debate, I think we shouldn't use the 'bog lava' rock walls on the western section of the lava area, by the bog merchant. That rock specifically has higher chances of gems, gold and silver. It also excluded copper and tin from being found, while providing standard iron and coal.

I'm happy to hear discussion on it; But I think we should have a section thats the old 'high' mineral chances with all types, while keeping the lava ravine 'unique' in having higher chances for the valuable ores.

Peter-Dolby commented:
1) Ah, so it -was- an intentional decision. Was never able to find a decision there before. The way I figured it was that a saiga isn't *that* difficult for the keep to acquire - it's like a hundred mammon in the Steward's. I'd always buy one as steward/duchess or just nick a spare from the Vanguard. But I guess the added friction of having to ask the steward is just enough to keep people from bothering ~90% of the time.
You sure we can't have just one? As a treat? The Stables can be very lonely otherwise...

2) I think we can have the old regular mineral chances in all the other parts of the map, but I feel for consistency's sake it makes most sense that you can assume that lava=x and not-lava=y, rather than lava indicating higher gem chances... some places but not other places.
Though now that I think about it, there's lava in the eastern mine, isn't there? And that doesn't have lava-type-ore-rarity? So maybe I was just overassuming based on the name

Peter-Dolby commented:
Few more little tinkery things.

A walk-in closet for the heirs, with a pouch for a smidge extra starting cash. Put the monkey statue on a table.
<img width="701" height="666" alt="78ece533ff93a9054aa2ecd08911fde4" src="https://github.com/user-attachments/assets/dd74a79d-8fe6-4bf5-b565-b4c3acc4c2f5" />


Tinkering with the secret Hand tunnel. Now requires a Hand key to get in or out, rather than giving anyone with a Walls key a secret route into the Hand's room. It's supposed to be secret! I assume!
<img width="1348" height="746" alt="bbd88fd603610dd6cd957a635e2ddaea" src="https://github.com/user-attachments/assets/b12e7555-7b3c-42b8-8d06-4745db22c241" />
Hand tunnel also has a little route back out behind the physician's. With a Hand's key.
<img width="996" height="477" alt="cf9a7167f7e6da03e99238c12690ca28" src="https://github.com/user-attachments/assets/16189eea-1d54-488e-a8dd-cadc13c5bb2e" />

Oh another change I forgot to mention - moving the Inn mailbox down closer to the paper and feathers rather than being tucked behind the bar where you have to clamber over the shelves to get to.
<img width="280" height="784" alt="3644330aa369afbc6ae9f980fec72738" src="https://github.com/user-attachments/assets/aa4b4691-5ad2-4529-baf5-b9350538f244" />

Peter-Dolby commented:
<img width="731" height="691" alt="1a6db8f3ea35dcea09f243d167f13bad" src="https://github.com/user-attachments/assets/cc4e47f5-7b18-4117-a8ea-877cebcf179a" />
Adding just a smidge of silk to the seamster's just to start them off - without going into the infinite never-ending lifetime supply that AP's map gives them

Peter-Dolby commented:
Noticed it's currently not TM'd - is it just the saigas and maybe the lava cave ore to adjust or anything else that needs changing?

Jamdrawers commented:
<img width="535" height="789" alt="image" src="https://github.com/user-attachments/assets/837d1872-69c0-419b-89e5-1346253e76fa" />
<img width="400" height="566" alt="image" src="https://github.com/user-attachments/assets/1755508a-663f-4bca-b81c-05df6d85cf57" />
Placing the entrance to the whitevein towards the center of town might make it more inviting, and feel less incongruous. It was always weird that the shady semi-legal drugs/gambling/prostitution den's entrance was pointed directly at the keep gate

Jamdrawers commented:
<img width="898" height="851" alt="image" src="https://github.com/user-attachments/assets/0748347e-31ad-41a7-b1b6-ec33a00418dc" />
I'd been learning more how to use custom Variables and without having to make any custom sprites I was able to make some pretty workable Curtains for use in the whitevein. I added a new room to the whitevein since I cut out the boring duplicate rooms.

<img width="719" height="574" alt="image" src="https://github.com/user-attachments/assets/001ec661-b5af-4b56-bfa0-94834aee87d5" />
Added curtains to the dressing area, felt awkward that the natural place to dress was in the corner of a large brightly lit room.

Jamdrawers commented:
<img width="846" height="534" alt="image" src="https://github.com/user-attachments/assets/4ad84209-f69c-49fa-b9a4-339f36434ad9" />
Okay nobody can stop me from tinkering with the brothel. Now that I understand how to use Variables properly - Room Keys! Now you can give a key to a patron so they can have a nap and get dressed without you having to baby them, or rent out a luxury room to two people to share without having to trust them with a key to the entire facility.

Later on I'd like to see the nightmaster have a special key for the nightmaster room that nightmaidens don't have but I'm afraid of touching that side of the code for now.

Jamdrawers commented:
A few more silly little needless changes.
1) A peddler in the lowtown foodshop. Lets the cheesemaker go off to get into hijinks without missing out on cheese sales. If the lowtown smith can have one then the baker should, too.
<img width="676" height="607" alt="8a91cc847a6480248dbe3edc26073a44" src="https://github.com/user-attachments/assets/90649cd2-03be-4c98-b71c-3599913c6b39" />
2) Some spare candles and foldable tables in manor storage near the kitchen so that the duchess can set up tables in the throne room or by the execution stands to host impromptu cake parties without need of a carpenter.
<img width="339" height="430" alt="8e3d68e494d1d78e2dd5bd8918868372" src="https://github.com/user-attachments/assets/ceccd46a-a888-4edd-bc39-d876e9c95b9c" />
3) Another go at the servant's rooms. I like little tented cubbyholes, offering a *bit* of privacy, but not *too much*. A little dressing-room area as well, you must be presentable for thy lord.
<img width="715" height="853" alt="bcf4bad37523d66b0431452e358148d6" src="https://github.com/user-attachments/assets/82bce6e5-9717-41b6-897b-fe51592694c1" />
4) Another pass at the armoury, just making it a little more interesting looking and organised.
<img width="680" height="465" alt="781b150a747e981aa89db8c4539b224e" src="https://github.com/user-attachments/assets/bdd8efc1-53e8-49d5-8e86-b52dbcac507c" />
5) The Outbound Boat. Anyone can go to the docks, get on a boat, and fuck off out of town, perhaps to escape an apocalypse. Just a copy of the Carriage, really.
<img width="1061" height="539" alt="550b0ad4eb6d271b1f1d912810a13579" src="https://github.com/user-attachments/assets/f9899015-4b21-443d-8fc0-f49766b671b3" />
6) Mountain Passes. Similarly, just a set of renamed carriage doors dotted around the edge of the map with refugee latejoin spawns clustered by them. Now you can leave the game via the bog, if such a thing is what your character would do. It's probably going to be less common to be used than the regular carriage but I think it's cute and makes sense-in-setting.
<img width="357" height="316" alt="d0cd427979df56e96150af36458f954a" src="https://github.com/user-attachments/assets/eeb5eeb0-6fda-4c49-bfa5-b5f03511030a" />

Jamdrawers commented:
Unless there are objections we can probably merge and if I think of anything to change then I can make another PR.
